cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/instance/AwsInstanceTemplate.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.cloudbreak.cloud.model.instance;
+
+import java.util.Collection;
+import java.util.Map;
+
+import com.sequenceiq.cloudbreak.cloud.model.InstanceStatus;
+import com.sequenceiq.cloudbreak.cloud.model.InstanceTemplate;
+import com.sequenceiq.cloudbreak.cloud.model.Volume;
+import com.sequenceiq.common.api.type.EncryptionType;
+
+public class AwsInstanceTemplate extends InstanceTemplate {
+
+    /**
+     * Key of the optional dynamic parameter denoting whether EBS encryption is enabled or not. This applies to both root & attached (data) volumes.
+     *
+     * <p>
+     *     Permitted values:
+     *     <ul>
+     *         <li>{@code Boolean.TRUE} instance, {@code "true"} (ignoring case): Encryption is enabled.</li>
+     *         <li>{@code Boolean.FALSE} instance, {@code "false"} (or any other {@code String} not equal to {@code "true"} ignoring case), {@code null}:
+     *         Encryption is disabled.</li>
+     *     </ul>
+     * </p>
+     *
+     * @see #putParameter(String, Object)
+     * @see Boolean#parseBoolean(String)
+     */
+    public static final String EBS_ENCRYPTION_ENABLED = "encrypted";
+
+    /**
+     * Key of the optional dynamic parameter denoting the type of AWS KMS encryption key to use for EBS encryption. Relevant only when encryption has been
+     * enabled via parameter {@link #EBS_ENCRYPTION_ENABLED}.
+     *
+     * <p>
+     *     Permitted values are the names of enum constants in {@link EncryptionType}:
+     *     <ul>
+     *         <li>{@code "NONE"}, {@code null}: No KMS key. Since EBS encryption always needs some KMS key, this setting essentially disables encryption.</li>
+     *         <li>{@code "DEFAULT"}: Use the <a href="https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#EBSEncryption_key_mgmt">default
+     *         key for EBS encryption</a> in the target Region. This can actually point to the AWS-managed KMS key with alias {@code alias/aws/ebs}, or
+     *         some customer-managed symmetric KMS key.</li>
+     *         <li>{@code "CUSTOM"}: Use the customer-managed symmetric KMS key specified in {@link #EBS_ENCRYPTION_KMS_KEY_ARN}.</li>
+     *     </ul>
+     * </p>
+     *
+     * @see #EBS_ENCRYPTION_ENABLED
+     * @see #EBS_ENCRYPTION_KMS_KEY_ARN
+     * @see #putParameter(String, Object)
+     * @see EncryptionType
+     */
+    public static final String EBS_ENCRYPTION_KMS_KEY_TYPE = "type";
```

Comment:
[As it is used in case of GCP](https://github.com/hortonworks/cloudbreak/pull/8340/files#diff-89018022889ad01df1fabcd052257e5bR78) we could move this constant one layer upper to the provider generic InstanceTemplate class.

Reply:
Done. Also moved the similarly common `"key"` constant.